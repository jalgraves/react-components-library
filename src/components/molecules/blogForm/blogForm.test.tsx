import React from "react"
import renderer from "react-test-renderer"
import BlogForm from "./blogForm"

it('blog form snapshot', () => {
  const component = renderer.create(
    <BlogForm 
      ariaDetails="TestBlogForm"
      buttonColor="red"
      buttonFontColor="white"
      buttonFontWeight="bold"
      buttonLetterSpacing=".3rem"
      createPostReply="This is a reply"
      display="flex"
      flexFlow="column wrap"
      fontFamily="Arial, Helvetica, sans-serif"
      invalidAuthorError="This is an invalid email error message"
      labelFontWeight="bold"
      labelLetterSpacing=".3rem"
      pFontColor="yellow"
      pFontFamily="Arial, Helvetica, sans-serif"
      pFontWeight="900"
      pLetterSpacing=".2rem"
      pTextTransform="uppercase"
      requiredFieldError="This is an error message"
      width="100%"
    />
  )
  let tree = component.toJSON()! as any
  console.log("BlogForm")
  console.log(tree.props)
  console.log(tree.children)
  expect(tree).toMatchSnapshot()
})
