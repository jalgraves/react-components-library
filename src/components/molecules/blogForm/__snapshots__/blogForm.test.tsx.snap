// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog form snapshot 1`] = `
<div
  aria-details="TestBlogForm"
  className="sc-beyTiQ jCiIGC"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Title
    </label>
    <input
      name="title"
      onBlur={[Function]}
      onChange={[Function]}
    />
    <label>
      Author
    </label>
    <input
      name="author"
      onBlur={[Function]}
      onChange={[Function]}
    />
    <label>
      Summary
    </label>
    <textarea
      name="summary"
      onBlur={[Function]}
      onChange={[Function]}
      rows={20}
    />
    <label
      htmlFor="tags"
    >
      Tags
    </label>
    <div
      aria-details="blog tags"
    >
      <div
        className="tagCheckBox"
      >
        <h3>
          foo
        </h3>
        <input
          name="tags"
          onBlur={[Function]}
          onChange={[Function]}
          type="checkbox"
          value="foo"
        />
      </div>
      <div
        className="tagCheckBox"
      >
        <h3>
          bar
        </h3>
        <input
          name="tags"
          onBlur={[Function]}
          onChange={[Function]}
          type="checkbox"
          value="bar"
        />
      </div>
    </div>
    <label>
      Post
    </label>
    <textarea
      name="body"
      onBlur={[Function]}
      onChange={[Function]}
      rows={20}
    />
    <input
      type="submit"
    />
  </form>
</div>
`;
